developer_manual/digging_deeper/task_processing.rst

```diff
@@ -38,6 +38,24 @@ The following built-in task types are available:
        * ``history``: ``ListOfTexts``
     * Output shape:
        * ``output``: ``Text``
+ * ``'core:text2text:chatwithtools'``: This task allows chatting with the language model with tools calling support. It is implemented by ``\OCP\TaskProcessing\TaskTypes\TextToTextChatWithTools``
+    * Input shape:
+       * ``system_prompt``: ``Text``
+       * ``input``: ``Text``
+       * ``history``: ``ListOfTexts``
+       * ``tools``: ``Text`` This should be a JSON object formatted like the one expected by the OpenAI API: https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools
```

Comment:
The tools parameter should be a JSON object formatted according to the OpenAI API specifications: https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools

Reply:
:+1: done